sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -200,15 +200,19 @@ class FrameLessOffsetWindowFunctionFrame(
 
   override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
     resetStates(rows)
-    if (ignoreNulls) {
-      findNextRowWithNonNullInput()
+    if (Math.abs(offset) > rows.length) {
+      inputIndex = offset
```

Comment:
what does this do? skip everything and return default value?

Reply:
yes, it does indirectly. To avoid confusion I modified change to limit the scope to `ignoreNulls` case only. Please take a look